azimuth/app.py

```diff
@@ -31,6 +43,38 @@
 _ready_flag: Optional[Event] = None
 
 
+COMMON_HTTP_ERROR_CODES = (
+    HTTP_400_BAD_REQUEST,
+    HTTP_401_UNAUTHORIZED,
+    HTTP_403_FORBIDDEN,
+    HTTP_404_NOT_FOUND,
+    HTTP_422_UNPROCESSABLE_ENTITY,  # Override responses from FastAPI ValidationError
+    HTTP_500_INTERNAL_SERVER_ERROR,
+    HTTP_503_SERVICE_UNAVAILABLE,
+)
+
+
+class HTTPExceptionModel(BaseModel):
+    detail: str
+
+
+async def handle_validation_error(request: Request, exception: RequestValidationError):
+    return JSONResponse(
+        status_code=HTTP_404_NOT_FOUND  # for errors in paths, e.g., /dataset_splits/potato
+        if "path" in (error["loc"][0] for error in exception.errors())
+        else HTTP_400_BAD_REQUEST,  # for other errors like in query params, e.g., pipeline_index=-1
+        content={"detail": str(exception)},
+    )
```

Comment:
I am not sure the `{ "detail": "..." }` JSON object wrapping the error message is useful. I could instead return a plain text error message.
```suggestion
    return PlainTextResponse(
        status_code=HTTP_404_NOT_FOUND  # for errors in paths, e.g., /dataset_splits/potato
        if "path" in (error["loc"][0] for error in exception.errors())
        else HTTP_400_BAD_REQUEST,  # for other errors like in query params, e.g., pipeline_index=-1
        content=str(exception),
    )
```
I would simply add an exception handler for `HTTPException` that does the same thing. I could then remove the 500 exception handler, as it was already returning plain text.
However, FastAPI's `http_exception_handler()` for `HTTPException` is a little bit more sophisticated as it supports setting response headers, so I thought it was cool to keep it as is.
Let me know if you have an opinion on this.